Content.Server/Radiation/Systems/RadiationSystem.cs

```diff
@@ -1,52 +1,48 @@
 ﻿using Content.Shared.Radiation.Events;
+using Robust.Shared.Configuration;
 using Robust.Shared.Map;
+using Robust.Shared.Physics.Systems;
 
 namespace Content.Server.Radiation.Systems;
 
-public sealed class RadiationSystem : EntitySystem
+public sealed partial class RadiationSystem : EntitySystem
 {
-    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly IMapManager _mapManager = default!;
+    [Dependency] private readonly SharedPhysicsSystem _physicsSystem = default!;
+    [Dependency] private readonly IConfigurationManager _cfg = default!;
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
 
-    private const float RadiationCooldown = 1.0f;
     private float _accumulator;
 
-    public override void Update(float frameTime)
+    public override void Initialize()
     {
-        base.Update(frameTime);
-        _accumulator += frameTime;
+        base.Initialize();
+        SubscribeCvars();
+        InitRadBlocking();
+    }
 
-        while (_accumulator > RadiationCooldown)
-        {
-            _accumulator -= RadiationCooldown;
-
-            // All code here runs effectively every RadiationCooldown seconds, so use that as the "frame time".
-            foreach (var comp in EntityManager.EntityQuery<RadiationSourceComponent>())
-            {
-                var ent = comp.Owner;
-                if (Deleted(ent))
-                    continue;
-
-                var cords = Transform(ent).MapPosition;
-                IrradiateRange(cords, comp.Range, comp.RadsPerSecond, RadiationCooldown);
-            }
-        }
+    public override void Shutdown()
+    {
+        base.Shutdown();
+        UnsubscribeCvars();
     }
 
-    public void IrradiateRange(MapCoordinates coordinates, float range, float radsPerSecond, float time)
+    public override void Update(float frameTime)
     {
-        var lookUp = _lookup.GetEntitiesInRange(coordinates, range);
-        foreach (var uid in lookUp)
-        {
-            if (Deleted(uid))
-                continue;
-
-            IrradiateEntity(uid, radsPerSecond, time);
-        }
+        base.Update(frameTime);
+
+        _accumulator += frameTime;
+        if (_accumulator < GridcastUpdateRate)
+            return;
```

Comment:
This should be stored on the component itself as it will lead to desyncs with pausing. Also you should -= for accumulators as otherwise you get float imprecision issues.

Reply:
Not sure if I understand. Do you mean store accumulator in radiation source components? Bit I need to process all radiation sources in one batch.

Reply:
How it's done now means if we ever snapshotted the server and reloaded for a replay would mean that it would cause desyncs because there's stateful data on the system.

Reply:
And what is current solution in codebase to store data that isn't attached to any entity like this system accumulator?